contracts/libraries/UniswapV2OracleLibrary.sol

```diff
@@ -0,0 +1,40 @@
+pragma solidity >=0.5.0;
+
+import '@uniswap/v2-core/contracts/interfaces/IUniswapV2Pair.sol';
+import '@uniswap/lib/contracts/libraries/FixedPoint.sol';
+
+// library with helper methods for oracles that are concerned with computing average prices
+library UniswapV2OracleLibrary {
+    using FixedPoint for *;
+
+    // helper function that returns the current block timestamp within the range of uint32, i.e. [0, 2**32 - 1]
+    function currentBlockTimestamp() internal view returns (uint32) {
+        return uint32(block.timestamp % 2 ** 32);
+    }
+
+    // helper function that returns the time that has elapsed since the given block timestamp in uint32
+    function timeElapsedSince(uint32 blockTimestampLast) internal view returns (uint32) {
+        // overflow desired
+        return currentBlockTimestamp() - blockTimestampLast;
+    }
+
+    // produces the cumulative price using counterfactuals to save gas and avoid a call to sync.
+    function currentCumulativePrices(
+        IUniswapV2Pair pair
```

Comment:
imo this should be a simple address

Reply:
i don't feel strongly about it, but why? it seems like type safety is good here

Reply:
it just feels like we're forcing an interface onto the consumers of the library that they may not want. if the function is meant to be called from an uncontrolled context it feels strange for the type of any of the arguments to not be native.